# Bike-share analysis case study

## Overview
The purpose of this project is to give insight of Bike-share activity at Cyclistic, A bike-share program that features more than 5,800 bicycles and 600 docking stations. The goal of this analysis is to found out casual riders and annual members use Cyclistic bikes differently and invent new strategy to convert casual riders into annual members. This dataset contain data from the customer of Cyclistic in span of one month. 


## Business Understanding
In 2016, Cyclistic launched a successful bike-share offering. The bikes can be unlocked from one station and returned to any other station in the system anytime.
Until now, Cyclistic’s marketing strategy relied on building general awareness and appealing to broad consumer segments. One approach that helped make these things possible was the flexibility of its pricing plans: single-ride passes, full-day passes, and annual memberships. Customers who purchase single-ride or full-day passes are referred to as casual riders. Customers who purchase annual memberships are Cyclistic members.

To formulate the analysis formulation there are couple of aspects needed to be covered:
- The Problem   : The marketing department wants to build a better campaign that could convert casual rider to annual member. This will be the key for the future growth of the company. since annual members are much more profitable than casual riders. 
- Data          : The data sourced sourced from https://divvy-tripdata.s3.amazonaws.com/index.html
- Objective     : Find the determinant variables and how annual members and casual riders use bikes differently so that it could help Casual rider to convert to annual member.
- Action        : Create a summary of the analysis and give recommendations based on the analysis.
- Value         :  This analysiswill help the company to make more profitsince annual members are much more profitable than casual riders. 

In [1]:
import numpy as np 
import pandas as pd 

In [2]:
activity = pd.read_csv('tripdata_2.csv')
activity

ride_id  rideable_type           started_at  \
0      89E7AA6C29227EFF   classic_bike  2021-02-12 16:14:56   
1      0FEFDE2603568365   classic_bike  2021-02-14 17:52:38   
2      E6159D746B2DBB91  electric_bike  2021-02-09 19:10:18   
3      B32D3199F1C2E75B   classic_bike  2021-02-02 17:49:41   
4      83E463F23575F4BF  electric_bike  2021-02-23 15:07:23   
...                 ...            ...                  ...   
49617  7ED482EE6C9F5DB2   classic_bike  2021-02-20 15:25:32   
49618  203DF22F090C1777   classic_bike  2021-02-09 08:54:38   
49619  940161523673F056    docked_bike  2021-02-27 14:46:06   
49620  C5538FFA492A7B00   classic_bike  2021-02-09 11:44:17   
49621  EB4CA525B953E123  electric_bike  2021-02-04 10:26:44   

                  ended_at         start_station_name start_station_id  \
0      2021-02-12 16:21:43   Glenwood Ave & Touhy Ave              525   
1      2021-02-14 18:12:09   Glenwood Ave & Touhy Ave              525   
2      2021-02-09 19:19:10         Clark St & Lake St     KA1503000012   
3      2021-02-02 17:54:06      Wood St & Chicago Ave              637   
4      2021-02-23 15:22:37         State St & 33rd St            13216   
...                    ...                        ...              ...   
49617  2021-02-20 15:59:45   Western Ave & Leland Ave     TA1307000140   
49618  2021-02-09 09:08:19  Franklin St & Chicago Ave            13017   
49619  2021-02-27 15:00:49  Franklin St & Chicago Ave            13017   
49620  2021-02-09 11:46:13  Franklin St & Chicago Ave            13017   
49621  2021-02-04 10:31:21  Franklin St & Chicago Ave            13017   

                 end_station_name end_station_id  start_lat  start_lng  \
0      Sheridan Rd & Columbia Ave            660  42.012701 -87.666058   
1        Bosworth Ave & Howard St          16806  42.012701 -87.666058   
2          State St & Randolph St   TA1305000029  41.885795 -87.631101   
3         Honore St & Division St   TA1305000034  41.895634 -87.672069   
4           Emerald Ave & 31st St   TA1309000055  41.834733 -87.625827   
...                           ...            ...        ...        ...   
49617                         NaN            NaN  41.966495 -87.688419   
49618                         NaN            NaN  41.896747 -87.635668   
49619       Michigan Ave & Oak St          13042  41.896747 -87.635668   
49620        Wells St & Walton St   TA1306000011  41.896747 -87.635668   
49621                         NaN            NaN  41.896743 -87.635678   

         end_lat    end_lng member_casual  
0      42.004583 -87.661406        member  
1      42.019537 -87.669563        casual  
2      41.884866 -87.627498        member  
3      41.903119 -87.673935        member  
4      41.838163 -87.645123        member  
...          ...        ...           ...  
49617  41.950000 -87.660000        member  
49618        NaN        NaN        member  
49619  41.900960 -87.623777        casual  
49620  41.899930 -87.634430        member  
49621  41.900000 -87.650000        member  

[49622 rows x 13 columns]

In [3]:
activity = activity.dropna()
activity2 = activity.copy()

In [4]:
activity2['started_at'] = activity2['started_at'].astype(np.datetime64)
activity2['ended_at'] = activity2['ended_at'].astype(np.datetime64)
activity2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 42996 entries, 0 to 49620
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   ride_id             42996 non-null  object        
 1   rideable_type       42996 non-null  object        
 2   started_at          42996 non-null  datetime64[ns]
 3   ended_at            42996 non-null  datetime64[ns]
 4   start_station_name  42996 non-null  object        
 5   start_station_id    42996 non-null  object        
 6   end_station_name    42996 non-null  object        
 7   end_station_id      42996 non-null  object        
 8   start_lat           42996 non-null  float64       
 9   start_lng           42996 non-null  float64       
 10  end_lat             42996 non-null  float64       
 11  end_lng             42996 non-null  float64       
 12  member_casual       42996 non-null  object        
dtypes: datetime64[ns](2), float64(4), object(7)
me

In [5]:
activity2['ride_length(minute)'] = (activity2['ended_at'] - activity2['started_at']).astype('timedelta64[m]')

In [6]:
cols = activity2.columns.tolist()
cols

['ride_id',
 'rideable_type',
 'started_at',
 'ended_at',
 'start_station_name',
 'start_station_id',
 'end_station_name',
 'end_station_id',
 'start_lat',
 'start_lng',
 'end_lat',
 'end_lng',
 'member_casual',
 'ride_length(minute)']

In [7]:
activity2 = activity2[['ride_id',
 'rideable_type',
 'started_at',
 'ended_at',
 'ride_length(minute)',
 'start_station_name',
 'start_station_id',
 'end_station_name',
 'end_station_id',
 'start_lat',
 'start_lng',
 'end_lat',
 'end_lng',
 'member_casual']]


In [8]:
activity2['day_of_week'] = activity2['started_at'].dt.day_name()
activity2

ride_id  rideable_type          started_at  \
0      89E7AA6C29227EFF   classic_bike 2021-02-12 16:14:56   
1      0FEFDE2603568365   classic_bike 2021-02-14 17:52:38   
2      E6159D746B2DBB91  electric_bike 2021-02-09 19:10:18   
3      B32D3199F1C2E75B   classic_bike 2021-02-02 17:49:41   
4      83E463F23575F4BF  electric_bike 2021-02-23 15:07:23   
...                 ...            ...                 ...   
49608  B883DC64BFA8D34E   classic_bike 2021-02-27 23:47:30   
49612  933F61AEAC149167   classic_bike 2021-02-12 08:09:33   
49614  1B94F316B15C79E0    docked_bike 2021-02-27 21:07:45   
49619  940161523673F056    docked_bike 2021-02-27 14:46:06   
49620  C5538FFA492A7B00   classic_bike 2021-02-09 11:44:17   

                 ended_at  ride_length(minute)  \
0     2021-02-12 16:21:43                  6.0   
1     2021-02-14 18:12:09                 19.0   
2     2021-02-09 19:19:10                  8.0   
3     2021-02-02 17:54:06                  4.0   
4     2021-02-23 15:22:37                 15.0   
...                   ...                  ...   
49608 2021-02-27 23:56:40                  9.0   
49612 2021-02-12 08:27:05                 17.0   
49614 2021-03-05 15:11:45               8284.0   
49619 2021-02-27 15:00:49                 14.0   
49620 2021-02-09 11:46:13                  1.0   

                   start_station_name start_station_id  \
0            Glenwood Ave & Touhy Ave              525   
1            Glenwood Ave & Touhy Ave              525   
2                  Clark St & Lake St     KA1503000012   
3               Wood St & Chicago Ave              637   
4                  State St & 33rd St            13216   
...                               ...              ...   
49608  Lincoln Ave & Belle Plaine Ave     TA1309000026   
49612    Milwaukee Ave & Wabansia Ave            13243   
49614       Franklin St & Chicago Ave            13017   
49619       Franklin St & Chicago Ave            13017   
49620       Franklin St & Chicago Ave            13017   

                 end_station_name end_station_id  start_lat  start_lng  \
0      Sheridan Rd & Columbia Ave            660  42.012701 -87.666058   
1        Bosworth Ave & Howard St          16806  42.012701 -87.666058   
2          State St & Randolph St   TA1305000029  41.885795 -87.631101   
3         Honore St & Division St   TA1305000034  41.895634 -87.672069   
4           Emerald Ave & 31st St   TA1309000055  41.834733 -87.625827   
...                           ...            ...        ...        ...   
49608         Clark St & Grace St   TA1307000127  41.956004 -87.680161   
49612  Larrabee St & Armitage Ave   TA1309000006  41.912616 -87.681391   
49614       Clark St & Leland Ave   TA1309000014  41.896747 -87.635668   
49619       Michigan Ave & Oak St          13042  41.896747 -87.635668   
49620        Wells St & Walton St   TA1306000011  41.896747 -87.635668   

         end_lat    end_lng member_casual day_of_week  
0      42.004583 -87.661406        member      Friday  
1      42.019537 -87.669563        casual      Sunday  
2      41.884866 -87.627498        member     Tuesday  
3      41.903119 -87.673935        member     Tuesday  
4      41.838163 -87.645123        member     Tuesday  
...          ...        ...           ...         ...  
49608  41.950780 -87.659172        member    Saturday  
49612  41.918084 -87.643749        member      Friday  
49614  41.967096 -87.667429        casual    Saturday  
49619  41.900960 -87.623777        casual    Saturday  
49620  41.899930 -87.634430        member     Tuesday  

[42996 rows x 15 columns]

In [9]:
activity2['day_of_week'].mode()

0    Saturday
dtype: object

In [10]:
cats = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
activity2['day_of_week'] = pd.Categorical(activity2['day_of_week'], categories=cats, ordered=True)
activity2 = activity2.sort_values('day_of_week')

In [11]:
activity2.describe()

ride_length(minute)     start_lat     start_lng       end_lat  \
count         42996.000000  42996.000000  42996.000000  42996.000000   
mean             20.772235     41.900492    -87.643103     41.900694   
std             232.417240      0.041556      0.023006      0.041564   
min               0.000000     41.651913    -87.774704     41.651895   
25%               6.000000     41.881320    -87.654787     41.881320   
50%              11.000000     41.894722    -87.639904     41.894877   
75%              19.000000     41.925330    -87.629138     41.925563   
max           30129.000000     42.064854    -87.534814     42.064854   

            end_lng  
count  42996.000000  
mean     -87.643334  
std        0.023030  
min      -87.774704  
25%      -87.654787  
50%      -87.640552  
75%      -87.629318  
max      -87.534814

In [12]:
means = activity2['ride_length(minute)'].groupby(activity2['member_casual']).mean()
means

member_casual
casual    46.646697
member    14.290638
Name: ride_length(minute), dtype: float64

In [13]:
day = activity2['ride_length(minute)'].groupby([activity2['member_casual'], activity2['day_of_week']]).mean()
day

member_casual  day_of_week
casual         Monday         30.856828
               Tuesday        42.908982
               Wednesday      27.926566
               Thursday       21.703088
               Friday         61.543745
               Saturday       60.721653
               Sunday         35.470685
member         Monday         14.519897
               Tuesday        14.474900
               Wednesday      13.742103
               Thursday       12.873301
               Friday         13.165963
               Saturday       15.869904
               Sunday         16.087018
Name: ride_length(minute), dtype: float64

In [14]:
day2 = activity2['ride_length(minute)'].groupby([activity2['member_casual'], activity2['day_of_week']])
day2.count()

member_casual  day_of_week
casual         Monday          454
               Tuesday         835
               Wednesday       926
               Thursday        842
               Friday         1223
               Saturday       3122
               Sunday         1211
member         Monday         3493
               Tuesday        5020
               Wednesday      5793
               Thursday       5225
               Friday         5682
               Saturday       5642
               Sunday         3528
Name: ride_length(minute), dtype: int64

In [15]:
import plotly.express as px
fig = px.histogram(
    activity2, 
    x='day_of_week', 
    color='member_casual',
    opacity= 0.5
    )
fig.show()

Based on the chart above, it can be seen that for annual members, wednesday is the most active with a total of 5793 riders per day on avarage. While the casual members are the most active on saturday with 3122 riders per day on avarage. overall, the day with the most bikes rent by rider happen on saturday, this probably due to saturday being a weekend so many people choose cycling as their holiday activity. 


Casual riders most active during this day probably due to holiday and casual riders only have time for cycling during their day off. While annual members mostly active during the weekday, this is probably because annual members use bicycle as a mean  of transportaion to their workplace.

In [16]:
d5 = day.to_frame()
d6 = d5.reset_index()
d6.sort_values(by = 'ride_length(minute)')

member_casual day_of_week  ride_length(minute)
10        member    Thursday            12.873301
11        member      Friday            13.165963
9         member   Wednesday            13.742103
8         member     Tuesday            14.474900
7         member      Monday            14.519897
12        member    Saturday            15.869904
13        member      Sunday            16.087018
3         casual    Thursday            21.703088
2         casual   Wednesday            27.926566
0         casual      Monday            30.856828
6         casual      Sunday            35.470685
1         casual     Tuesday            42.908982
5         casual    Saturday            60.721653
4         casual      Friday            61.543745

In [17]:
fig = px.bar(
    d6, 
    x='day_of_week',
    y = 'ride_length(minute)',
    color = 'member_casual',
    color_discrete_map= {'member' : 'blue',
    'casual': 'red'},
    opacity= 0.5
    )
fig.show()

Based on the chart above, it can be seen that on avarage, casual riders spent more time on the ride than annual members even on a weekday. This probably caused by the travel speed. Because casual riders who do not cycling regulary is slower than anual members who cycling frequently. For casual biker the ride length is peak on friday at avarage of 61 minute.
While for member biker is peak at sunday where they can cycling at ease because its weekend at avarage of 16 minutes.

## Suggestion
Based from Exploratory Data Analysis, there are few suggestion for Cyclistic to tailor their marketing campaign for converting casual riders into members: 

- The difference between annual members and casual riders is that annual members are active mostly during the weekday This probably becaused the annual members use bike as means of commute to their workplace. While for casual riders, they are active mostly during the weekend. this probably because casual riders didn't want to or cannot use bike as their means of transportation to the workplace and dont have time to cycling after they get home from work.

-  What differ casual rider and annual member how much they cycling during weekday. Casual rider will likely buy annual membership if they are cycling frequently for commute and during weekday

- Cyclistic can use digital media to influence casual riders to become members by advertised it on friday, because on that day, the amount of casual rider reach its highest number during the weekday and to convert casual rider to annual member is best to do on weekday. Since casual rider that cycling during weekend likely people who don't have time during the weekday therefore only cycling occasionally and harder to convert to annual member